galsim/utilities.py

```diff
@@ -330,34 +349,137 @@ def thin_tabulated_values(x, f, rel_err=1.e-4, preserve_range=False):
         # Nothing to do
         return x,f
 
+    total_integ = np.trapz(abs(f), x)
+    thresh = total_integ * rel_err
+
+    if not preserve_range:
+        errleft = 0.5 * (abs(f[1])+abs(f[0])) * (x[1]-x[0])
+        errright = 0.5 * (abs(f[-1])+abs(f[-2])) * (x[-1]-x[-2])
+        accumulated_err = 0.0
+        # arbitrarily allocate half of error budget to trimming the endpoints.
+        # while at least one endpoint can be trimmed:
```

Comment:
Very interesting.  I think you could be right about the blue-to-red point.  I had worried about that.  In particular at sharp features (like emission lines or breaks) it might often get right up next to the feature on one segment and have the next point (across the feature) induce too big an error to continue.  But then for the segment that includes the break say, it might be ok with going a couple of points beyond the break before the error is too large.  So this could systematically skew the shape toward the blue.

In Jim's test, we aren't thinning the seds, just the bandpasses, but the same principal would hold at atmospheric absorption lines and such.  They might be being skewed systematically toward the blue by the old algorithm.  And I wouldn't be surprised if there were more such features in the u band than the others.

Anyway, I completely concur with keeping the new algorithm and use preserve_range=True as the default.  I'd just like to try to speed it up a bit if we can...  :)